src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -660,23 +653,28 @@ private function addService($id, Definition $definition)
             $lazyInitialization = '';
         }
 
-        // with proxies, for 5.3.3 compatibility, the getter must be public to be accessible to the initializer
-        $isProxyCandidate = $this->getProxyDumper()->isProxyCandidate($definition);
-        $visibility = $isProxyCandidate ? 'public' : ($definition->isPublic() ? 'protected' : 'private');
+        $asFile =  $this->asFiles && $definition->isShared();
```

Comment:
to not impact them because they can be is a hot path loop, non-shared definitions still use methods